src/java.base/share/native/launcher/main.c

```diff
@@ -278,15 +321,32 @@ main(int argc, char **argv)
         margv = args->elements;
     }
 
-    // Avoid unexpected process completion when checkpointing under docker container run
-    // by creating the main process waiting for children before exit.
-    if (is_checkpoint && 1 == getpid()) {
-        g_child_pid = fork();
-        if (0 < g_child_pid) {
-            // The main process should forward signals to the child.
-            setup_sighandler();
-            const int status = wait_for_children();
-            exit(status);
+    if (is_checkpoint) {
+        const int crac_min_pid_default = 128;
+        const char *env_min_pid_str = getenv("CRAC_MIN_PID");
```

Comment:
Why using an environment variable rather than JVM option? Env vars cannot be 'listed', so the user can't know about them without finding a specific place in documentation.

Reply:
This is because of PID adjustment  happens before JVM in its, so to use JVM option we need to have JVM option parsing implemented in main.c (at least for one option), more complicated than it was done in parse_checkpoint(). That is why I've decided to use env var instead. OTOH maybe it worth considering moving PID adjustment & forking implementation to java.c, closer to JVM_Init call.

Reply:
Implemented -XX:CRaCMinPid option